pkg/operator/ceph/object/user/controller.go

```diff
@@ -0,0 +1,229 @@
+	}
+}
+
+// TODO: Need to think through how to do updates, leaving out for now.
+
+// func (c *ObjectStoreUserController) onUpdate(oldObj, newObj interface{}) {
+// 	// if the object store user spec is modified, update the object store user
+// 	oldStoreUser, err := getObjectStoreUserObject(oldObj)
+// 	if err != nil {
+// 		logger.Errorf("failed to get old objectstoreuser object: %+v", err)
+// 		return
+// 	}
+// 	newStoreUser, err := getObjectStoreUserObject(newObj)
+// 	if err != nil {
+// 		logger.Errorf("failed to get new objectstoreuser object: %+v", err)
+// 		return
+// 	}
+//
+// 	if !storeUserChanged(oldStoreUser.Spec, newStoreUser.Spec) {
+// 		logger.Debugf("object store user %s did not change", newStoreUser.Name)
+// 		return
+// 	}
+//
+// 	logger.Infof("applying object store user %s changes", newStoreUser.Name)
+// 	if err = UpdateStoreUser(c.context, *newStoreUser, c.rookImage, c.hostNetwork, c.storeUserOwners(newStoreUser)); err != nil {
+// 		logger.Errorf("failed to create (modify) object store user %s. %+v", newStoreUser.Name, err)
+// 	}
+// }
+
+func (c *ObjectStoreUserController) onDelete(obj interface{}) {
+	user, err := getObjectStoreUserObject(obj)
+	if err != nil {
+		logger.Errorf("failed to get objectstoreuser object: %+v", err)
+		return
+	}
+
+	if err = deleteUser(c.context, user); err != nil {
+		logger.Errorf("failed to delete object store user %s. %+v", user.Name, err)
+	}
+}
+
+func (c *ObjectStoreUserController) storeUserOwners(store *cephv1beta1.ObjectStoreUser) []metav1.OwnerReference {
+	// Only set the cluster crd as the owner of the object store user resources.
+	// If the object store user crd is deleted, the operator will explicitly remove the object store user resources.
+	// If the object store user crd still exists when the cluster crd is deleted, this will make sure the object store user
+	// resources are cleaned up.
+	return []metav1.OwnerReference{c.ownerRef}
+}
+
+func getObjectStoreUserObject(obj interface{}) (objectstoreuser *cephv1beta1.ObjectStoreUser, err error) {
+	var ok bool
+	objectstoreuser, ok = obj.(*cephv1beta1.ObjectStoreUser)
+	if ok {
+		// the objectstoreuser object is of the latest type, simply return it
+		return objectstoreuser.DeepCopy(), nil
+	}
+	return nil, fmt.Errorf("not a known objectstoreuser object: %+v", obj)
+}
+
+// Create the user
+func (c *ObjectStoreUserController) createUser(context *clusterd.Context, u *cephv1beta1.ObjectStoreUser) error {
+	// validate the user settings
+	if err := ValidateUser(context, u); err != nil {
+		return fmt.Errorf("invalid user %s arguments. %+v", u.Name, err)
+	}
+
+	// create the user
+	logger.Infof("creating user %s in namespace %s", u.Name, u.Namespace)
+	userConfig := cephrgw.ObjectUser{
+		UserID:      u.Name,
+		DisplayName: &u.Name,
+	}
+	objContext := cephrgw.NewContext(context, u.Spec.Store, u.Namespace)
+
+	user, rgwerr, err := cephrgw.CreateUser(objContext, userConfig)
```

Comment:
What happens if someone creates a user through `radosgw-admin` and then creates a user with the same name in the crd? We don't necessarily need to support that case, but I'm curious if we get back the same creds for the existing user and it will just work.

Reply:
I tested this scenario just now. If the user gets created with radosgw-admin and then a crd gets created with the same name, the access-key and secret-key for the existing user will be stored in the k8s secret. So it appears to just work.